xmake/rules/nodejs/module/xmake.lua

```diff
@@ -0,0 +1,90 @@
+--!A cross-platform build utility based on Lua
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- Copyright (C) 2015-present, TBOOX Open Source Group.
+--
+-- @author      ruki
+-- @file        xmake.lua
+--
+
+rule("nodejs.module")
+    on_load(function(target)
+        -- imports
+        import("core.cache.detectcache")
+        import("core.project.target", { alias = "project_target" })
+
+        -- set kind
+        if target:is_plat("macosx") then
+            target:set("kind", "binary")
+            target:add("ldflags", "-bundle", "-undefined dynamic_lookup", { force = true })
+        else
+            target:set("kind", "shared")
+        end
+
+        -- export symbols
+        if target:is_plat("windows") then
+            target:add("shflags", "/export:napi_register_module_v1", "/export:node_api_module_get_api_version_v1",
+                { force = true })
+        else
+            target:set("symbols", "none")
+        end
+
+        -- https://github.com/nodejs/node-addon-api/issues/1021
+        if is_plat("mingw") then
+            import("core.project.config")
+            local outputdir = path.join(target:autogendir(), "/node-api-stub")
+            if not os.isdir(outputdir) then
+                import("devel.git")
+                git.clone("https://github.com/napi-bindings/node-api-stub", { depth = 1, outputdir = outputdir })
```

Comment:
we should move it to xmake-repo, and use `add_packages` + target:pkg("xx") to get it.

Reply:
Refer https://github.com/xmake-io/xmake-repo/pull/6004